shared/util/quit-helper.desktop.js

```diff
@@ -3,18 +3,21 @@ import {ipcRenderer, ipcMain, app} from 'electron'
 import {quit} from '../../desktop/app/ctl'
 import {hideDockIcon} from '../../desktop/app/dock-icon'
 
-export type Context = {type: 'uiWindow'} | {type: 'mainThread'} | {type: 'quitButton'}
+export type Context = 'uiWindow' | 'mainThread' | 'quitButton' | 'beforeQuit'
+export type Action = 'closePopups' | 'quitMainWindow' | 'quitApp'
```

Comment:
Same argument here, although having extra fields here may be more important (akin to flux actions)